imageregistry/v1/types_imagepruner.go

```diff
@@ -0,0 +1,86 @@
+package v1
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImagePrunerList is a slice of ImagePruner objects.
+type ImagePrunerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []ImagePruner `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImagePruner is the configuration object for an image registry pruner
+// managed by the registry operator.
+type ImagePruner struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec ImagePrunerSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// +optional
+	Status ImagePrunerStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+// ImagePrunerSpec defines the specs for the running image pruner.
+type ImagePrunerSpec struct {
+	// schedule specifies when to execute the job.
+	// Uses standard cronjob syntax: https://wikipedia.org/wiki/Cron
+	// Defaults to 0 0 * * *
+	// +optional
+	Schedule string `json:"schedule" protobuf:"bytes,1,opt,name=schedule"`
+	// suspend specifies whether or not to suspend subsequent executions of this cronjob.
+	// Defaults to false
+	// +optional
+	Suspend *bool `json:"suspend,omitempty" protobuf:"bytes,2,opt,name=suspend"`
+	// keepTagRevisions specifies how many tag revisions to keep.
+	// Defaults to 5
```

Comment:
Update to align with our `oc` help text:

```
// keepTagRevisions specifies the number of image revisions for a tag in an image stream that will be preserved.
```

Reply:
Also period at the end: `Defaults to 5.`

Reply:
updated in current push